libs/nucs/src/policy.rs

```
@@ -0,0 +1,316 @@
+    }
+
+    fn build_simple_operator<'de, A>(self, mut seq: A, op: String) -> Result<OperatorPolicy, A::Error>
+    where
+        A: SeqAccess<'de>,
+    {
+        let selector = self.build_selector(&mut seq, 1)?;
+        let value = seq.next_element::<serde_json::Value>()?.ok_or_else(|| Error::invalid_length(2, &"a value"))?;
+        let operator = match op.as_str() {
+            "==" => Operator::Equals(value),
+            "!=" => Operator::NotEquals(value),
+            _ => panic!("unsupported operator"),
+        };
+        Ok(OperatorPolicy { selector, operator })
+    }
+
+    fn build_list_operator<'de, A>(self, mut seq: A, op: String) -> Result<OperatorPolicy, A::Error>
+    where
+        A: SeqAccess<'de>,
+    {
+        let selector = self.build_selector(&mut seq, 1)?;
+        let values = seq
+            .next_element::<Vec<serde_json::Value>>()?
+            .ok_or_else(|| Error::invalid_length(2, &"a list of values"))?;
+        let operator = match op.as_str() {
+            "anyOf" => Operator::AnyOf(values),
+            _ => panic!("unsupported operator"),
+        };
+        Ok(OperatorPolicy { selector, operator })
+    }
+
+    fn build_simple_connector<'de, A>(self, mut seq: A, connector: String) -> Result<ConnectorPolicy, A::Error>
+    where
+        A: SeqAccess<'de>,
+    {
+        let policy = seq.next_element::<Box<Policy>>()?.ok_or_else(|| Error::invalid_length(1, &"a policy"))?;
+        let connector = match connector.as_str() {
+            "not" => ConnectorPolicy::Not(policy),
+            _ => panic!("unsupported operator"),
+        };
+        Ok(connector)
+    }
+
+    fn build_list_connector<'de, A>(self, mut seq: A, connector: String) -> Result<ConnectorPolicy, A::Error>
+    where
+        A: SeqAccess<'de>,
+    {
+        let policies =
+            seq.next_element::<Vec<Policy>>()?.ok_or_else(|| Error::invalid_length(1, &"a list of policies"))?;
```

Just to confirm, you truly want to return an error in cases like `[ "and", [] ]` ?

---

This is up for discussion, I'm on the fence on this one. Maybe allowing this will give you more flexibility when building more complex conditions.